spec/migration/parity_check/dynamic_request_content_spec.rb

```diff
@@ -501,6 +501,161 @@
+      let(:identifier) { :teacher_api_id_for_change_schedule_action }
+      let(:ecf_induction_record1) { FactoryBot.create(:migration_induction_record) }
+      let(:ecf_participant_profile1) { ecf_induction_record1.participant_profile }
+      let(:ecf_induction_record2) { FactoryBot.create(:migration_induction_record) }
+      let(:ecf_participant_profile2) { ecf_induction_record2.participant_profile }
+      let!(:ecf_induction_record3) { FactoryBot.create(:migration_induction_record) }
+
+      let(:ecf_cpd_lead_provider) { FactoryBot.create(:migration_cpd_lead_provider) }
+      let(:ecf_lead_provider) { FactoryBot.create(:migration_lead_provider, id: lead_provider.ecf_id, cpd_lead_provider: ecf_cpd_lead_provider) }
+      let!(:ecf_partnership) { FactoryBot.create(:migration_partnership, lead_provider: ecf_lead_provider) }
+
+      let!(:ecf_participant_declaration) { FactoryBot.create(:migration_participant_declaration, cpd_lead_provider: ecf_cpd_lead_provider, participant_profile: ecf_participant_profile1) }
+
+      before do
+        ecf_induction_record1.induction_programme.update!(partnership: ecf_partnership)
+        ecf_induction_record2.induction_programme.update!(partnership: ecf_partnership)
+      end
+
+      it { is_expected.to eq(ecf_induction_record2.preferred_identity.external_identifier) }
+    end
+
+    context "when fetching `change_schedule_different_schedule_and_cohort_participant_body`" do
+      let(:identifier) { :change_schedule_different_schedule_and_cohort_participant_body }
+      let(:lead_provider_delivery_partnership) { FactoryBot.create(:lead_provider_delivery_partnership, active_lead_provider:) }
+      let(:school_partnership) { FactoryBot.create(:school_partnership, active_lead_provider:) }
+      let!(:training_period) { FactoryBot.create(:training_period, :for_ect, :ongoing, school_partnership:) }
+      let(:teacher) { training_period.trainee.teacher }
+
+      let(:contract_period_2022) { FactoryBot.create(:contract_period, year: 2022) }
+      let(:active_lead_provider_2022) { FactoryBot.create(:active_lead_provider, lead_provider:, contract_period: contract_period_2022) }
+
+      before do
+        # Create 2022 contract period with partnership
+        FactoryBot.create(:lead_provider_delivery_partnership, active_lead_provider: active_lead_provider_2022)
+
+        # Deferred participant for current lead provider
+        FactoryBot.create(:training_period, :for_ect, :ongoing, :deferred, school_partnership:)
+        # Withdrawn participant for current lead provider
+        FactoryBot.create(:training_period, :for_ect, :ongoing, :withdrawn, school_partnership:)
+        # Participants for different lead providers should not be used
+        FactoryBot.create(:training_period, :for_ect, :ongoing)
+
+        allow(Schedule).to receive(:identifiers).and_return({ "ecf-extended-january" => "ecf-extended-january", "ecf-replacement-september" => "ecf-replacement-september" })
```

Comment:
Is it worth using `ecf-extended-january` for the new option and `training_period.schedule.identifier` for the second to make it clearer that we're changing to a new schedule? same for `change_schedule_different_schedule_participant_body`